src/libs/_tests/runtime-utils.test.js

```diff
@@ -179,11 +226,24 @@ describe('getCurrentPersistentDisk', () => {
     expect(getCurrentPersistentDisk(tools.galaxy.appType, [cromwellProvisioning], [cromwellProvisioningDisk])).toBeUndefined()
   })
   it('returns the newest attached disk, even if app is deleting', () => {
-    expect(getCurrentPersistentDisk(tools.galaxy.appType, mockApps, mockAppDisks)).toBe(galaxyDeletingDisk)
-    expect(getCurrentPersistentDisk(tools.cromwell.appType, mockApps, mockAppDisks)).toBe(cromwellProvisioningDisk)
+    expect(getCurrentPersistentDisk(tools.galaxy.appType, mockApps, mockAppDisks, 'test-workspace')).toBe(galaxyDeletingDisk)
+    expect(getCurrentPersistentDisk(tools.cromwell.appType, mockApps, mockAppDisks, 'test-workspace')).toBe(cromwellProvisioningDisk)
   })
   it('returns the newest unattached disk that is not deleting if no app instance exists', () => {
-    expect(getCurrentPersistentDisk(tools.galaxy.appType, [], mockAppDisks)).toBe(galaxyDisk)
-    expect(getCurrentPersistentDisk(tools.cromwell.appType, [galaxyRunning], mockAppDisks)).toBe(cromwellUnattachedDisk)
+    expect(getCurrentPersistentDisk(tools.galaxy.appType, [], mockAppDisks, 'test-workspace')).toBe(galaxyDisk)
+    expect(getCurrentPersistentDisk(tools.cromwell.appType, [galaxyRunning], mockAppDisks, 'test-workspace')).toBe(cromwellUnattachedDisk)
+  })
+  it('returns a galaxy disk only if it is in the same workspace as the previous app it was attached to', () => {
+    expect(getCurrentPersistentDisk(tools.galaxy.appType, [], mockAppDisks, 'test-workspace')).toBe(galaxyDisk)
+    expect(getCurrentPersistentDisk(tools.galaxy.appType, [], mockAppDisks, 'incorrect-workspace')).toBeUndefined()
+  })
+})
+
+describe('projectHasMultipleDisks', () => {
+  it('returns true when there are multiple disks', () => {
+    expect(projectHasMultipleDisks(mockAppDisks + galaxyDisk3)).toBe(true)
```

Comment:
Note this is confusing because mockAppDisks contains disks related to different app types.

I really would be tempted to have `workspaceHasMultipleDisks` do the app filtering itself (so you would pass in the appType as a variable) because I think it makes the method much easier to understand.

Reply:
I like the idea of changing the method to do the app filtering, i'll update!